### DISCUSSION TITLE
`workspace_swipe` action from a keyboard bind?

### DISCUSSION THREAD

**Discussion by Ferk** [hyprwm/Hyprland#1450] [Q&A]
I'd like to set up a way to switch to the next/prev workspace within a display, in a similar way as how it's possible to switch to the next/prev window within a workspace (using `cyclenext`). Even better if it creates a new workspace in the display when going past the last one.

From what I understand, there's already a feature that does more or less what I wanted, but it seems to be exclusively reserved for gestures (for touchpad/touchscreen, I believe?), `workspace_swipe`.

I do not use a touchpad or touchscreen but I would like to be able to trigger this action with the keyboard (or maybe with a combination of $Mod+mousewheel). Is there a way to do it?

**Comment by lplawlor** [comment]
+1 To this. The best I've been able to come up with using binds is:

```nix
      bind = [
        # ...
        
        # Focus workspace (on same monitor)
        "SUPER CTRL, left, workspace, r-1"
        "SUPER CTRL, right, workspace, r+1"

        # Move focused window to workspace (on same monitor)
        "SUPER CTRL SHIFT, left, movetoworkspace, r-1"
        "SUPER CTRL SHIFT, right, movetoworkspace, r+1"
        
        #...
      ];
```

However this comes with the drawback of creating any number of new workspaces. When using the swipe gesture, you're able to create just one new workspace at the end of line, and then no more if that one stays empty. With these binds, I can keep pressing `SUPER CTRL right` and keep creating higher and higher workspace IDs. Then to get back, you have to do the same with `SUPER CTRL left`.

Alternatively, you can use `m` instead of `r`, but then you cannot create any new empty workspaces.

I'm not familiar enough with Hyprland's codebase to know the best way to allow for this behaviour, but perhaps a new workspace identifier like `m` or `r` that mimics the behaviour of the workspace swipe gesture? Or maybe a new dispatcher altogether?

**Comment by gabbieblue** [comment]
this solution was very helpful to me, i just want to point out that you can use r when going to the next workspace, allowing new ones to be created, and m to go to the previous workspace, skipping creating new ones while going backwards which is undesirable behaviour
```
        # workspace swipe but with $mod+scroll
        "$mod, mouse_up, workspace, r+1"
        "$mod, left, workspace, m-1"
        "$mod, mouse_down, workspace, m-1"
        "$mod, right, workspace, r+1"
        # send window to workspace
        "$mod shift, mouse_up, movetoworkspace, r+1"
        "$mod shift, left, movetoworkspace, m-1"
        "$mod shift, mouse_down, movetoworkspace, m-1"
        "$mod shift, right, movetoworkspace, r+1"
```
this still creates workspaces even when the current one is empty, but aside from having a different ID, this is functionally the same as the swipe gesture I believe
hope this helps someone!

**Comment by code-IM-perfect** [comment]
Would like to see this implemented too

Relevant Issue: #6915 (it was force closed in the great purge)
